main.cpp

```diff
@@ -240,46 +240,87 @@ static int add_endpoint_address(const char *name, size_t name_len, const char *i
     return ret;
 }
 
+static int add_bauds(unsigned long **bauds, size_t *bauds_size, const char *bauds_str)
+{
+    char *tmp, *nul_bauds_str;
+    int count = 1; // we expect at least one baud
+    int i, ret;
+
+    if (!bauds_str || bauds_str[0] == '\0') {
+        *bauds = (unsigned long *)calloc(sizeof(unsigned long), 1);
+        *bauds[0] = DEFAULT_BAUDRATE;
+        *bauds_size = 1;
+        return 0;
+    }
+
+    nul_bauds_str = strdup(bauds_str);
+    if (!nul_bauds_str) {
+        return -errno;
+    }
+
+    for (tmp = nul_bauds_str; *tmp != '\0'; tmp++) {
+        if (*tmp == ',') {
+            count++;
+            *tmp = '\0';
+        }
+    }
+
+    *bauds = (unsigned long *)calloc(sizeof(unsigned long), count);
+    if (!*bauds) {
+        ret = -errno;
+        goto bauds_error;
+    }
+
+    i = 0;
+    *bauds_size = count;
+    tmp = nul_bauds_str;
+    while (count--) {
+        if (safe_atoul(tmp, &(*bauds)[i]) < 0) {
+            log_error("Invalid baud %s", tmp);
+            ret = -EINVAL;
+            goto atoul_error;
+        }
+        i++;
+        tmp += strlen(tmp) + 1;
+    }
+
+    free(nul_bauds_str);
```

Comment:
any reason why we are not using strtok? This looks complicated for something that will execute once.